sdk/iot/Azure.Iot.Hub.Service/src/API Design.md

```diff
@@ -107,7 +107,56 @@ APIs for managing configurations for devices and modules.
 <details>
 
 ```csharp
+/// <summary>
+/// Gets a configuration on the IoT Hub for automatic device/module management
+/// </summary>
+/// <param name="configurationId">The unique identifier of the configuration.</param>
+/// <param name="cancellationToken">The cancellation token.</param>
+/// <returns>The retrieved automatic device/module management twin configuration</returns>
+public virtual async Task<Response<TwinConfiguration>> GetConfigurationAsync(string configurationId, CancellationToken cancellationToken = default)
+
+/// <summary>
+/// Create or update a configuration on the IoT Hub for automatic device/module management
+/// </summary>
+/// <param name="configuration">Twin configuration to update</param>
+/// <param name="precondition">The condition on which to perform this operation</param>
+/// <param name="cancellationToken">The cancellation token.</param>
+/// <returns>The created automatic device/module management twin configuration</returns>
+public virtual async Task<Response<TwinConfiguration>> CreateOrUpdateConfigurationAsync(TwinConfiguration configuration, IfMatchPrecondition precondition = IfMatchPrecondition.IfMatch, CancellationToken cancellationToken = default)
+
+/// <summary>
+/// Deletes a configuration on the IoT Hub for automatic device/module management
+/// </summary>
+/// <param name="configurationId">The unique identifier of the configuration.</param>
+/// <param name="precondition">The condition on which to perform this operation</param>
+/// <param name="cancellationToken">The cancellation token.</param>
+/// <returns>The http response <see cref="Response{T}"/>.</returns>
+public virtual async Task<Response> DeleteConfigurationAsync(string configurationId, IfMatchPrecondition precondition = IfMatchPrecondition.IfMatch, CancellationToken cancellationToken = default)
```

Comment:
I'm not sure that I follow how this API will work. If I pass in precondition = ifMatch, what value am I matching against the service representation? Shouldn't this API take the full configuration object with the ETag embedded in it like we did for the delete device identity API?

Reply:
That's a good point, I'll change this to twin configuration. It's surprising that this wasn't noticed in our review meeting